### ISSUE TITLE
Need a better story wrt logging

### ISSUE THREAD

**Issue by glandium** [mozilla/sccache#151]
There are essentially two logging setups, and they don't work the same way.
- If you set `RUST_LOG`, you end up with the env_logger setup, which allows fine-grained log output (I've found `RUST_LOG=sccache::compiler=debug` to be incredibly useful, so much so that I'm thinking we should set it by default on Firefox builds, as well as `SCCACHE_ERROR_LOG` and upload the log as artifact. BUT, it has no timestamps.
- If you instead set `SCCACHE_LOG_LEVEL`, you can, AIUI, only set the global log level (and `debug` is already too verbose), BUT you get timestamps (although you don't get fractions of seconds, which makes the timestamps less useful for measures). OTOH, you also always get the log sent to either stdout or stderr, I'm not sure which, on the client side (which doesn't follow `SCCACHE_ERROR_LOG`), and that breaks most build systems (for Firefox, configure just plain fails)

**Comment by luser** [comment]
Agreed. It's sort of a mess. `RUST_LOG` + `SCCACHE_ERROR_LOG` is currently the best you can do. Not having the fine-grained log level filtering makes the logging unusable due to other crates having verbose logging. @alexcrichton has a PR that I have not found time to look at that might help: https://github.com/mozilla/sccache/pull/115

If you want to review that that would be appreciated!

More generally: we currently pull in both `env_logger` and `fern`, and the latter doesn't support the fine-grained log level syntax that the former does. I thought about extracting those bits into a standalone crate to make things nicer, but never quite got motivated.
